From 7f862961cc5294eec23aa0d0840f9e9e5c60f1f6 Mon Sep 17 00:00:00 2001
From: Adrian Alonso <adrian.alonso@nxp.com>
Date: Thu, 28 Apr 2016 14:00:11 -0500
Subject: [PATCH 1030/5242] MLK-12732: ARM: imx: imx6q: lppdr2 fix wrong
 timings for lower freq operation

commit  6a0cb6082c41127eb3244f979e93a0141caf2b7a from
https://source.codeaurora.org/external/imx/linux-imx.git

On low frequency operation (freq <= 100Mhz) set self-refresh exit
to next valid command delay to 23 clock cycles (MMDC_MCFG0[tXS] = 0x16)

Signed-off-by: Adrian Alonso <adrian.alonso@nxp.com>
Signed-off-by: Ranjani Vaidyanathan <ranjani.vaidyanathan@nxp.com>
(cherry-picked from commit b480c0a99fe722017e1ad04c1de16739c1467e0a)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-imx/lpddr2_freq_imx6q.S |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-imx/lpddr2_freq_imx6q.S b/arch/arm/mach-imx/lpddr2_freq_imx6q.S
index 1eb83a4..3eede66 100644
--- a/arch/arm/mach-imx/lpddr2_freq_imx6q.S
+++ b/arch/arm/mach-imx/lpddr2_freq_imx6q.S
@@ -236,7 +236,7 @@ wait_div_update2:
 	str	r6, [r8, #0x18]
 
 	/* Adjust LPDDR2 timings for 24Mhz operation */
-	ldr	r5, =0x03032073
+	ldr	r5, =0x03162073
 	str	r5, [r8, #0xC]  	/* MMDC0_MDCFG0 */
 	ldr	r7, =0x00020482
 	str	r7, [r8, #0x10] 	/* MMDC0_MDCFG1 */
-- 
1.7.9.5

